dashboard_description_plugin/src/dp_dashboard_description_plugin/hooks/useAttributes.ts

```diff
@@ -0,0 +1,71 @@
+// (C) 2023 GoodData Corporation
+
+import { useEffect, useMemo, useState } from "react";
+import { IdentifierRef, idRef } from "@gooddata/sdk-model";
+import { ICustomWidget, useWidgetFilters } from "@gooddata/sdk-ui-dashboard";
+import { IDataSlice, UseCancelablePromiseStatus, useInsightDataView } from "@gooddata/sdk-ui";
+
+import { AttributesMap, AttributesMapByInsight } from "../interface.js";
+import { TooManyDataPointsError } from "../utils.js";
+
+type UseAttributesConfig = {
+    attributes: AttributesMapByInsight | undefined;
+    widget: ICustomWidget;
+};
+type UseAttributes = (config: UseAttributesConfig) => {
+    attributesSlice: IDataSlice | null;
+    attributesError: Error | null;
+    attributesStatus: UseCancelablePromiseStatus;
+    attributesInsightId?: IdentifierRef;
+    attributeConfigs?: AttributesMap;
+};
+
+export const useAttributes: UseAttributes = ({ attributes, widget }) => {
```

Comment:
@dremileta can we add the short annotation about what this hook does (extracts attributes from an insight) here as well please, feel free to preprocess it ;)